_posts/2017/2017-11-01-internal-workshops.md

```diff
@@ -0,0 +1,197 @@
+
+You can find the content of the workshop, including the slides, [on GitHub](https://github.com/thokari/gradle-workshop).
+
+
+## Structure of the workshop
+
+This was the agenda of the workshop.
+
+- Motivation
+- Evolution of build tools
+- Language basics
+- Gradle concepts
+- Building Java projects
+- Gradle and IDEs
+
+The workshop starts with a short introduction and explanation of the motivation to
+create and use a tool like Gradle, as well as my own personal motivation to work
+with, and teach others about it.
+
+Many ideas in Gradle are easier to explain and understand when put in the context
+of the history of build tools. This is especially true, if someone migrated from
+[Make](https://www.gnu.org/software/make/) directly to Gradle, for example, and was
+wondering [why no one was using make for Java](https://stackoverflow.com/questions/2209827/why-is-no-one-using-make-for-java).
+
+Gradle is primarily used to build Java projects.
+However, I find it important to cover fundamentals of the tool first, before looking
+at the Java-specific things.<br>
+Gradle makes extensive use of a [Groovy](http://groovy-lang.org/) DSL (domain specific
+language), so it makes sense to cover the basics of that language.<br>
+Below that, Gradle offers a rich model for the domain of building software projects.
+
+When learning how Gradle is used in Java projects I find it very helpful to start
+with simple examples. For example, I prefer to look at everything as files on the
+file system, instead of whatever view your IDE has to offer. These are related of
+course, but different (I am looking at you, Eclipse workspace).<br>
+A big part of what Gradle tries to achieve is a unified view of the project, that
+is independent from any IDE-specific configuration or presentation.
+
+This is why I only touch on IDEs in the end, and create the examples and exercises
+in a way they do not require any IDE to work with them.
+
+
+## Personal Motivation
+
+When I was learning about [Gradle](https://gradle.org/) 6 years ago, I found the
+tool quite compelling. Having used [Apache Ant](http://ant.apache.org/) before and
+getting increasingly annoyed by it, Gradle promised exactly what I was looking for -
+a way to refactor and reorganize build logic, by being able to make sense of it.
+
+Back then, many of my colleagues seemed to be reluctant to use Gradle, or did not
+buy into the philosophy behind it.
+For example, I remember someone being specifically proud of using the same Ant `build.xml`
+since ten years.
+
+At the same time I was fortunate to get the opportunity to participate in a workshop
+by Gradleware, which I found very inspiring. The combination of these things gave
+me the idea to condense and adapt the content of this workshop.
+
+When I joined ePages, Gradle was used quite extensively here already. There are
+several Gradle multi-projects in our current system (e.g. REST API, GUI testing
```

Comment:
The examples you've put (REST API, GUI testing...), can any of these be linked?

Reply:
Unfortunately not...